packages/apptive_grid_user_management/lib/src/delete_account.dart

```diff
@@ -0,0 +1,209 @@
+    final key = GlobalKey<DeleteAccountState>();
+    return DeleteAccount._(
+      key: key,
+      onAccountDeleted: onAccountDeleted,
+      child: Builder(
+        builder: (context) {
+          return ListTile(
+            onTap: () => key.currentState?.showConfirmationDialog(context),
+            title: Text(
+              ApptiveGridUserManagementLocalization.of(context)!.deleteAccount,
+              style: TextStyle(color: Theme.of(context).colorScheme.error),
+            ),
+          );
+        },
+      ),
+    );
+  }
+
+  /// A [TextButton] that on tap will show a confirmation dialog to delete the user's account
+  /// [onAccountDeleted] will be called if the user's account was deleted successfully
+  factory DeleteAccount.textButton({
+    required void Function() onAccountDeleted,
+  }) {
+    final key = GlobalKey<DeleteAccountState>();
+    return DeleteAccount._(
+      key: key,
+      onAccountDeleted: onAccountDeleted,
+      child: Builder(
+        builder: (context) {
+          return TextButton(
+            onPressed: () => key.currentState?.showConfirmationDialog(context),
+            child: Text(
+              ApptiveGridUserManagementLocalization.of(context)!.deleteAccount,
+              style: TextStyle(
+                color: Theme.of(context).colorScheme.error,
+              ),
+            ),
+          );
+        },
+      ),
+    );
+  }
+
+  final Widget _child;
+
+  /// A callback invoked when the user's account was deleted
+  final void Function() onAccountDeleted;
+
+  @override
+  State<DeleteAccount> createState() => DeleteAccountState();
+}
+
+/// The State for [DeleteAccount]
+class DeleteAccountState extends State<DeleteAccount> {
+  @override
+  Widget build(BuildContext context) {
+    return widget._child;
+  }
+
+  /// Shows an [AlertDialog] to have the user confirm that they want their account deleted
+  void showConfirmationDialog(BuildContext context) {
+    final localization = ApptiveGridUserManagementLocalization.of(context);
+    final client = ApptiveGridUserManagement.maybeOf(context)?.client;
+    final loadingKey = GlobalKey<_LoadingStateWidgetState>();
+    showDialog(
+      barrierDismissible: false,
+      context: context,
+      builder: (BuildContext dialogContext) {
+        return _LoadingStateWidget(
+          key: loadingKey,
+          child: Builder(
+            builder: (context) {
+              return AlertDialog(
+                title: Text(localization?.deleteAccount ?? 'Delete Account'),
+                content: Column(
+                  mainAxisSize: MainAxisSize.min,
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: [
+                    Text(
+                      localization?.deleteAccountConfirmation ??
+                          'Are you sure you want to delete your account?\n This can not be undone.',
+                    ),
+                    if (_LoadingStateWidget.error(context) != null)
```

Comment:
Maybe extract this into a variable, that makes it easier to read

Reply:
Changed